pkg/kwokctl/components/etcd.go

```diff
@@ -114,6 +116,10 @@ func BuildEtcdComponent(conf BuildEtcdComponentConfig) (component internalversio
 		)
 	}
 
+	if conf.Version.GTE(version.NewVersion(3, 4, 0)) && conf.LogLevel != log.InfoLevelSecurity {
+		etcdArgs = append(etcdArgs, "--log-level="+conf.LogLevel)
```

Comment:
https://etcd.io/docs/v3.5/upgrades/upgrade_3_4/#deprecating-etcd---debug-flag-now---log-leveldebug

If etcd is less than 3.4 and the logs level is debug, then we use `--debug`